lib/matplotlib/cbook/__init__.py

```diff
@@ -2009,7 +2007,7 @@ def _warn_external(message, category=None):
     etc.).
     """
     frame = sys._getframe()
-    for stacklevel in itertools.count(1):
+    for stacklevel in itertools.count(1):  # lgtm[py/unused-loop-variable]
```

Comment:
really, lgtm can't see that stacklevel is used below the loop?...

Reply:
Not using the loop variable within the loop is often an error, even if it's used afterwards. I think this is what they want to indicate here.

Reply:
Frankly, I don't like peppering the code with comments to appease bad linters.

Reply:
Well, I think that the alert is a valid concern, and a statement that `stacklevel` is intentionally not used in the loop is justified here. I did not put comments in other places where the linting was just not good enough.

Reply:
I guess I should just write something like (untested)
```
    def recur(f, x0):
        return itertools.accumulate(lambda x, _: f(x), itertools.repeat(x0))

    stacklevel = 1 + next(
        i for i, fr in enumerate(recur(lambda fr: fr.f_back, sys._getframe()))
        if not re.match(r"\A(matplotlib|mpl_toolkits)(\Z|\.)",
                        fr.f_globals["name"])
```
and keep lgtm happy :p
(abusing accumulate to repeatedly just apply a function is actually documented as an example in https://docs.python.org/3/library/itertools.html#itertools.accumulate...)